djstripe/models.py

```diff
@@ -174,91 +120,38 @@ def send_signal(self):
-    refund_gross = models.DecimalField(decimal_places=2, max_digits=7)
-    validation_count = models.IntegerField()
-    validation_fees = models.DecimalField(decimal_places=2, max_digits=7)
 
     objects = TransferManager()
 
-    def __str__(self):
-        return "<amount={amount}, status={status}, stripe_id={stripe_id}>".format(amount=self.amount, status=self.status, stripe_id=self.stripe_id)
-
-    def update_status(self):
-        self.status = self.api_retrieve().status
-        self.save()
-
     @classmethod
-    def process_transfer(cls, event, transfer):
-        defaults = {
-            "amount": transfer["amount"] / decimal.Decimal("100"),
-            "status": transfer["status"],
-            "date": convert_tstamp(transfer, "date"),
-            "description": transfer.get("description", ""),
-            "adjustment_count": transfer["summary"]["adjustment_count"],
-            "adjustment_fees": transfer["summary"]["adjustment_fees"],
-            "adjustment_gross": transfer["summary"]["adjustment_gross"],
-            "charge_count": transfer["summary"]["charge_count"],
-            "charge_fees": transfer["summary"]["charge_fees"],
-            "charge_gross": transfer["summary"]["charge_gross"],
-            "collected_fee_count": transfer["summary"]["collected_fee_count"],
-            "collected_fee_gross": transfer["summary"]["collected_fee_gross"],
-            "net": transfer["summary"]["net"] / decimal.Decimal("100"),
-            "refund_count": transfer["summary"]["refund_count"],
-            "refund_fees": transfer["summary"]["refund_fees"],
-            "refund_gross": transfer["summary"]["refund_gross"],
-            "validation_count": transfer["summary"]["validation_count"],
-            "validation_fees": transfer["summary"]["validation_fees"],
-        }
-        for field in defaults:
-            if field.endswith("fees") or field.endswith("gross"):
-                defaults[field] = defaults[field] / decimal.Decimal("100")
-
-        if event.kind == "transfer.paid":
-            defaults.update({"event": event})
-            obj, created = Transfer.objects.get_or_create(
-                stripe_id=transfer["id"],
-                defaults=defaults
-            )
-        else:
-            obj, created = Transfer.objects.get_or_create(
-                stripe_id=transfer["id"],
-                event=event,
-                defaults=defaults
-            )
+    def process_transfer(cls, event, stripe_object):
+        try:
+            transfer = cls.stripe_objects.get_by_json(stripe_object)
+            created = False
+        except cls.DoesNotExist:
+            transfer = cls.create_from_stripe_object(stripe_object)
+            created = True
+
+        transfer.event = event
 
         if created:
-            for fee in transfer["summary"]["charge_fee_details"]:
-                obj.charge_fee_details.create(
+            transfer.save()
```

Comment:
Seems like the transfer is going to get saved no matter what. Might as well make a call to save at the end of the method.